index.html

```diff
@@ -1887,324 +1887,639 @@ <h2>
+            <li>If <var>details</var>["<a>region</a>"] is present, set
+            <var>address</var>.<a>[[\region]]</a> to the result of trimming
+            <var>details</var>["<a>region</a>"]
+            </li>
+            <li>If <var>details</var>["<a>phone</a>"] is present:
+              <ol>
+                <li>Check if <var>details</var>["<a>phone</a>"] is a
+                <a>structurally valid phone number</a>, throw a
+                <a>RangeError</a>.
+                </li>
+                <li>Set <var>address</var>.<a>[[\phone]]</a> to the result of
+                <a>canonicalize a phone number</a>
+                <var>details</var>["<a>phone</a>"].
+                </li>
+              </ol>
+            </li>
+            <li>If <var>details</var>["<a>city</a>"] is present, set
+            <var>address</var>.<a>[[\city]]</a> to the result of trimming <var>
+              details</var>["<a>city</a>"].
+            </li>
+            <li>If <var>details</var>["<a>dependentLocality</a>"] is present,
+            set <var>address</var>.<a>[[\dependentLocality]]</a> to the result
+            of trimming <var>details</var>["<a>dependentLocality</a>"].
+            </li>
+            <li>If <var>details</var>["<a>postalCode</a>"] is present, set
+            <var>address</var>.<a>[[\postalCode]]</a> to the result of trimming
+            <var>details</var>["<a>postalCode</a>"].
+            </li>
+            <li>If <var>details</var>["<a>sortingCode</a>"] is present, set
+            <var>address</var>.<a>[[\sortingCode]]</a> to the result of
+            trimming <var>details</var>["<a>sortingCode</a>"].
+            </li>
+            <li>If <var>details</var>["<a>languageCode</a>"] is present:
+              <ol>
+                <li>If <a data-cite=
+                "ecma-402#sec-isstructurallyvalidlanguagetag">IsStructurallyValidLanguageTag</a>(<var>details</var>["<a>languageCode</a>"])
+                is false, throw a <a>RangeError</a> exception.
+                </li>
+                <li>Set <var>address</var>.<a>[[\languageCode]]</a> to
+                <a data-cite=
+                "ecma-402#sec-canonicalizelanguagetag">CanonicalizeLanguageTag</a>(<var>details</var>["<a>languageCode</a>"]).
+                </li>
+              </ol>
+            </li>
+            <li>If <var>details</var>["<a>organization</a>"] is present, set
+            <var>address</var>.<a>[[\organization]]</a> to the result of
+            trimming <var>details</var>["<a>organization</a>"].
+            </li>
+            <li>If <var>details</var>["<a>recipient</a>"] is present, set <var>
+              address</var>.<a>[[\recipient]]</a> to the result of trimming
+              <var>details</var>["<a>recipient</a>"].
+            </li>
+            <li>Return <var>address</var>.
+            </li>
+          </ol>
+        </section>
+        <section>
+          <h2>
+            Creating a <code>PaymentAddress</code>
```

Comment:
Pardon my noob question. Why do we need both "constructor" and "creating" sections?

Reply:
Yeah, we might need to merge them... one is supposed to impose order when contracting from user input (should rename this section more clearly), while the other is for calling the constructor from script.

Reply:
*constructing

Reply:
I'm going to change this algo to use an `AddressInit` dictionary instead. The algorithm will also take an optional "excludeList", which will exclude dictionary members as per #648. That should solve for `onpaymentaddress` privacy issues.